### PR TITLE
updatecheck demo - DO NOT COMMIT

### PR DESCRIPTION
**Pull request by decibel**
This is a demonstration of a working updatecheck target.

That target works by pulling down UPGRADE_FROM version of pgtap via pgxn (currently defaults to 0.94.0), installing the current version, and then doing

```
CREATE EXTENSION pgtap VERSION :'UPGRADE_FROM';
ALTER EXTENSION pgtap UPDATE :'CURRENT_VERSION';
```

then running the unit tests.

The other big change in here is that all installcheck activity is now managed with pg_regress execution schedules. There are two big wins from that:
- The extension is now loaded only once, not for every test
- Tests are now run in parallel by default (specify PARALLEL_CONN=1 to force serial)

To support that, there's now a test/build directory where temporary build objects are put.

If we're happy with this code then I'll get it into master and start hammering out the remaining upgrade issues.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,15 +1,19 @@
+# THERE IS ANOTHER .gitignore IN test/sql!
+
 .*.swp
 pgtap.sql
 pgtap-core.sql
 pgtap-schema.sql
 uninstall_pgtap.sql
-test/setup.sql
```
Comment (theory) [inline .gitignore]: Why is this deleted; it's  a generated file. Unless you mean to put it into `test/.gitignore`?

**Comment by decibel** [comment]
Where is it generated at? There's no target in the Makefile for it...

**Comment by theory** [comment]
Oh, I guess I switched to a static file in b8c7cb2c1f0432a354ff84fa5064eeae0c1942cf but neglected to remove it from `.gitignore`. So it's not generated anymore!

**Comment by nasbyj** [comment]
Testing of extension update was merged in #128.
